recipes/ggd/post-link.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
+set -eo pipefail -o nounset
+
+## Organize required conda channels
+conda config --add channels defaults
+conda config --add channels ggd-genomics
```

Comment:
If we're going to change people's channel order on them we had better advertise that we're doing so.

Reply:
We should not mess around with the channel order in post-link ... 
If you want to do these things print a message to your user with instructions but don't mess around with settings that are outside of the package dir please.

Reply:
We just discussed this internally within the core team and we're pretty uncomfortable with using post-link actions to modify people's channels. If adding `ggd-genomics` to the channels is really required then I suggest adding this package there instead.

Reply:
This gets a big fat **not approved** from my part.
It should *at least* be
```sh
\. "${PREFIX}"/etc/profile.d/conda.sh
conda activate "${PREFIX}"
conda config --env ...
```
to restrict it to the install environment.

But even then:
General advice: If anyhow possible, avoid post-link scripts. It's non-standard code that needs non-standard maintenance and may fail in non-standard ways. Overall, it's just costly.

Reply:
I note your concerns, however, the post-link script was deliberately set up this way to provide usability with "GGD" channels. The `gg-genomics` channel is added here as a requirement of GGD. However, we didn't want to force the "GGD" channels to take precedence over other channels such as conda-forge or bioconda. Additionally, it is recommended that conda-forge have the highest priority and bioconda the second highest. I am happy to put a message that informs the user that we are changing the channel priority, or even only adding GGD channels, but this is a vital part of the recipe. 

Additionally, the `ggd-genomics` and any other GGD channel is primarily for data packages and does not support software.

Reply:
I think it is fine if you give instructions and a nice readme which is displayed when people install this package, but please do not change "system" setting on behalf of the user.

Reply:
@bgruening, one of the advantages of using "conda" and the ideas of recipes is to provide dependency handling for the user so they do not have to do it themselves. This is the intent for this step of the process.

Reply:
We're aware of that, but this change isn't one we can permit in Bioconda recipes. This is a deal breaker for accepting the package.

Reply:
Is it allowable if add only the `ggd-genomics` channel?

Reply:
No, we can't allow changing people's channel orders on the system level. What you probably can do is change your internal usage of `conda` to inject it there (e.g., making system calls to `conda install -c ggd-genomics ...` or figuring out where in the conda code it reads in the channels and modify that). I imagine asking users to instead modify their channels is easier though.

Reply:
If we reach consensus on this PR to use a post-link script which `conda config` invocations, the points from my previous comment still stand:
1. should use "its own `conda`", i.e., the one in "${PREFIX}"
2. has to add `--env` since otherwise we are messing with users' configuration outside of the environment.

Reply:
BTW, there are several ways to "inject" additional channels for `conda` to consider. I don't know what the workflow with `ggd` is, but if the channel `ggd-genomic` is only queried thorough `ggd` and the user is not expected to necessarily interact with `conda` directly, it might be worthwhile to inject `ggd-genomic` during runtime. Feel free to ask/ping me on https://github.com/gogetdata/ggd-cli if you need pointers (I know one or two things about `conda`'s innards ;).

Reply:
This has been removed from the post-link script